src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
```

Comment:
checklist 2 [CODE STYLE]: Use switch statement if you have limited amount of conditions.